lizmap/modules/lizmap/lib/Form/QgisForm.php

```diff
@@ -1245,15 +1245,22 @@ private function fillControlFromValueRelationLayer($fieldName, $formControl)
         // In lib/jelix/plugins/formwidget/menulist_html/menulist_html.formwidget.php
         // An empty value is added with these rules
         // if ($this->ctrl->emptyItemLabel !== null || !$this->ctrl->required)
-        // In lizmap, we add an empty value except for required allowMulti
-        // Lizmap does not take into account the QGIS configuration $formControl->valueRelationData['allowNull']
-        // because of the way empty value is added by Jelix
+        // In lizmap, we add an empty value except for every ValueRelation without allowMulti
         $formControl->ctrl->emptyItemLabel = '';
-        if ($formControl->valueRelationData['allowMulti'] && $formControl->ctrl->required) {
-            $formControl->ctrl->emptyItemLabel = null;
-        }
+
+        // Create Datasource
         $dataSource = new QgisFormValueRelationDynamicDatasource($formControl->ref);
 
+        // In lib/jelix/plugins/formwidget/checkboxes_html/checkboxes_html.formwidget.php
+        // control emptyItemLabel or required is not tacking into account
```

Comment:
@rldhont i don't understand this sentence, do you mean:
```suggestion
        // control emptyItemLabel where required is not taken into account
```

Reply:
Lets say yes, fix typo layer if needed please @rldhont

Reply:
It was not `emptyItemLabel where required` but more `emptyItemLabel not null or required positif`